etc/compile_check.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+set -e # exit when any command fails
+
+GC=go
+WD=$(pwd)
+COMPILE_CHECK_DIR="internal/test/compilecheck"
+DEV_MIN_VERSION=1.19
+
+# version will flatten a version string of upto 4 components for inequality
+# comparison.
+function version {
+	echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }';
+}
+
+# dev_compile_check will attempt to build in the development environment. This
+# check will only run on environments where the Go version is greater than or
+# equal to the DEV_MIN_VERSION.
+function dev_compile_check {
+	MACHINE_VERSION=`${GC} version | { read _ _ v _; echo ${v#go}; }`
+
+	if [ $(version $MACHINE_VERSION) -lt $(version $DEV_MIN_VERSION) ]; then
+		return
+	fi
+
+	GO111MODULE=on ${GC} build ./...
+	GO111MODULE=on ${GC} build ${BUILD_TAGS} ./...
+}
+
+# compile_check will attemps to build the the internal/test/compilecheck project
+# using the provided go version. This is to simulate a end-to-end use case. This
+# check will only run on environments where the Go version is greater than or
+# equal to the given version.
+function compile_check {
+	GO111MODULE=$1
+	VERSION=$2
+	MACHINE_VERSION=`${GC} version | { read _ _ v _; echo ${v#go}; }`
+
+	if [ $(version $MACHINE_VERSION) -lt $(version $VERSION) ]; then
+		return
+	fi
```

Comment:
I'm concerned this logic could lead to builds that pass without actually running anything. For example, when we update the Evergreen CI Go version from 1.19 to 1.20, if we forget to also update this script, `compile_check` will return without actually building anything because the Go version doesn't match any of the expected versions. Additionally, we want to be able to run these scripts locally with any supported version of Go to debug issues.

Is there a way to just skip running `go mod tidy` on Go 1.13 without tying the script to specific Go versions?

Ideas:
Remove the `$VERSION` argument and update the `go mod tidy` block to:
```
if [ "$MACHINE_VERSION" != 1.13 ]; then
	go mod tidy
fi
```